host-controller/src/main/java/org/jboss/as/host/controller/HostControllerEnvironment.java

```diff
@@ -922,6 +923,22 @@ public OperationStepHandler getHostNameWriteHandler(LocalHostControllerInfoImpl
         return new HostNameWriteAttributeHandler(hostControllerInfo);
     }
 
+    void setStability(Stability stability) {
+        WildFlySecurityManager.setPropertyPrivileged(ProcessEnvironment.STABILITY, stability.toString());
+        this.hostSystemProperties.put(ProcessEnvironment.STABILITY, stability.toString());
+        this.stability = stability;
+    }
+
+    public void checkStabilityIsValidForInstallation(Stability stability) {
+        checkStabilityIsValidForInstallation(productConfig, stability);
+    }
+
+    private void checkStabilityIsValidForInstallation(ProductConfig productConfig, Stability stability) {
+        if (!productConfig.getStabilitySet().contains(stability)) {
+            throw HostControllerLogger.ROOT_LOGGER.unsupportedStability(this.stability, productConfig.getProductName());
+        }
+    }
```

Comment:
Notice we also have https://issues.redhat.com/browse/WFCORE-6760 to move common functionality to ProcessEnvironment, that can be done later